src/typeahead/highlight.spec.ts

```diff
@@ -28,153 +29,109 @@ function highlightHtml(fixture) {
-
-  it('should not convert null result to string', async(inject([TestComponentBuilder], (tcb) => {
-       const html = '<ngb-highlight [result]="null" term="null"></ngb-highlight>';
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('');
-           });
-     })));
-
-  it('should properly detect matches in 0 result', async(inject([TestComponentBuilder], (tcb) => {
-       const html = '<ngb-highlight [result]="0" term="0"></ngb-highlight>';
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe(`<span class="ngb-highlight">0</span>`);
-           });
-     })));
-
-  it('should not higlight anything for blank term', async(inject([TestComponentBuilder], (tcb) => {
-       const html = '<ngb-highlight result="1null23" [term]="null"></ngb-highlight>';
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('1null23');
-           });
-     })));
-
-  it('should not higlight anything for blank term', async(inject([TestComponentBuilder], (tcb) => {
-       const html = `<ngb-highlight result="123" [term]="''"></ngb-highlight>`;
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('123');
-           });
-     })));
-
-  it('should properly higlight zeros', async(inject([TestComponentBuilder], (tcb) => {
-       const html = `<ngb-highlight result="0123" [term]="0"></ngb-highlight>`;
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('<span class="ngb-highlight">0</span>123');
-           });
-     })));
-
-  it('should support custom highlight class', async(inject([TestComponentBuilder], (tcb) => {
-       const html = '<ngb-highlight result="123" [term]="2" highlightClass="my"></ngb-highlight>';
-       tcb.overrideTemplate(TestComponent, html)
-           .createAsync(TestComponent)
-           .then((fixture: ComponentFixture<TestComponent>) => {
-             fixture.detectChanges();
-             expect(highlightHtml(fixture)).toBe('1<span class="my">2</span>3');
-           });
-     })));
+  beforeEach(() => {
+    TestBed.overrideModule(NgbTypeaheadModule, {set: {exports: [NgbHighlight]}});
```

Comment:
Yes, otherwise all the tests fail. My understanding is that the test module is just like any regular module: it imports NgbTypeaheadModule, but NgbHighlight is not exported by NgbTypeaheadModule, so it's an internal component of NgbTypeaheadModule, that may not be used outside of NgbTypeaheadModule, including in the test module. 

Overriding the exports of NgbTypeaheadModule allows exporting the internal NgbHighlight component (i.e. do as if it was exported), and thus using it in the test module.